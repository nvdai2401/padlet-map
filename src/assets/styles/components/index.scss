@import './top-bar';
@import './folk-me';
@import './spinner';
@import './post-content';
@import './post-popup';
@import './mini-post';
@import './post-preview';
@import './post-expanded';
@import './map-control';
@import './map-view';
