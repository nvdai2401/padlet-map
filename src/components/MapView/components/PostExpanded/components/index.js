export { default as PostSlider } from './PostSlider';
