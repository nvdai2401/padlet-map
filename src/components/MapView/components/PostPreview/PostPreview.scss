.post-preview {
  display: flex;
  z-index: zIndex('post-preview');
  background-color: rgba(43, 43, 43, 0.8);
  backdrop-filter: blur(10px);
  @include size($width: 100vw);
  @include absolute($left: 0, $bottom: 0);

  .post-count {
    display: none;
  }

  .post-list {
    overflow: auto;

    .post-list__container {
      .divider {
        margin: 16px 0;
        border-left: 1px solid #333333;
      }
    }
  }

  @include desktop {
    left: unset;
    flex-direction: column;
    padding-bottom: 16px;
    border-radius: 6px;
    @include absolute($right: 12px, $bottom: 72px);
    @include size($width: 320px, $maxHeight: calc(100% - 96px));

    .post-list {
      flex-direction: column;

      .post-list__container {
        flex-direction: column;
        .divider {
          margin: 0 16px;
          border-left: unset;
          border-top: 1px solid #333333;
        }
      }
    }

    .post-count {
      display: inline-block;
      color: hsla(0, 0%, 100%, 0.7);
      font-weight: 600;
      padding: 12px 16px;
      text-align: left;
      font-size: 0.875rem;
    }
  }
}
