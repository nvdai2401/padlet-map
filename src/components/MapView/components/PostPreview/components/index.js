export { default as MiniPost } from './MiniPost';
