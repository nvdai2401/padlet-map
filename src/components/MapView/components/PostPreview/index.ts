export { default } from './PostPreview.vue';
