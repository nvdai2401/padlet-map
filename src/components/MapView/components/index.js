export { default as Post } from './Post';
export { default as PostPreview } from './PostPreview';
export { default as PostExpanded } from './PostExpanded';
