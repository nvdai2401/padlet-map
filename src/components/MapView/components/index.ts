export { default as PostContent } from './PostContent';
export { default as PostPopup } from './PostPopup';
export { default as PostPreview } from './PostPreview';
export { default as PostExpanded } from './PostExpanded';
export { default as MapControl } from './MapControl';
