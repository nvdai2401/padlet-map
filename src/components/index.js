export { default as MapView } from './MapView';
export { default as TopBar } from './TopBar';
