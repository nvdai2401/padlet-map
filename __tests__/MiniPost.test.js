/* eslint-disable no-undef */
import { posts } from './shares/mockData';
import { shallowMountWithProps } from './shares/utils';
import { MiniPost } from '@/components/MapView/components/PostPreview/components';

const propsData = {
  post: posts[0],
};

describe('<MiniPost />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMountWithProps(MiniPost, propsData);
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it(`should contain post title`, () => {
    expect(wrapper.find('.title').text()).toBe(propsData.post.headline);
  });

  it(`should contain post subtitle`, () => {
    expect(wrapper.find('.subtitle').exists()).toBeTruthy();
  });

  it(`should contain post image`, () => {
    expect(wrapper.find('img').exists()).toBeTruthy();
  });
});
