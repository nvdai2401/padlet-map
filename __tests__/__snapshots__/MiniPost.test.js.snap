// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MiniPost /> should match snapshot 1`] = `
<div
  class="flex-row justify-space-between mini-post"
>
  <div
    class="flex-column post-title"
  >
    <span
      class="title"
      style="color: red;"
    >
      United States
    </span>
     
    <span
      class="subtitle"
    >
      <div>
        Came here to study in 2006.
      </div>
    </span>
  </div>
   
  <div
    class="post-image"
  >
    <img
      alt="United States"
      height="56px"
      loading="lazy"
      src="https://ik.imagekit.io/fireman35/pics/usa_98nvVlwOEN.jpg?tr=w-112"
      width="56px"
    />
  </div>
</div>
`;
